FlowCrypt/Controllers/Setup/SetupGenerateKeyViewController.swift

```diff
@@ -128,32 +127,46 @@ private actor Service {
             keyDetails: [encryptedPrv.key],
             passPhrase: storageMethod == .persistent ? passPhrase: nil,
             source: .generated,
-            for: user.email
+            for: userId.email
         )
     }
 
+    // todo - there is a similar method in EnterpriseServierApi
+    //   this should be put somewhere general
+    private func getIdToken(for user: User) async throws -> String? {
+        switch user.authType {
+        case .oAuthGmail:
+            return try await GoogleUserService(
+                currentUserEmail: user.email,
+                appDelegateGoogleSessionContainer: nil // needed only when signing in/out
+            ).getCachedOrRefreshedIdToken()
+        default:
+            return Imap(user: user).imapSess?.oAuth2Token
+        }
+    }
```

Comment:
To file another issue - extract `getIdToken` methods in `SetupGenerateKeyViewController` and `EnterpriseServerApi`

Reply:
https://github.com/FlowCrypt/flowcrypt-ios/issues/1367